packages/store/src/Bytes.sol

```diff
@@ -0,0 +1,705 @@
+
+  /**
+   * Converts a `bytes` memory array to a single `bytes` memory value.
+   * TODO: optimize gas cost
+   */
+  function from(bytes[] memory input) internal pure returns (bytes memory output) {
+    output = new bytes(0);
+    for (uint256 i; i < input.length; i++) {
+      output = bytes.concat(output, input[i]);
+    }
+  }
+
+  function from(uint8[] memory input) internal pure returns (bytes memory output) {
+    bytes32 ptr;
+    assembly {
+      ptr := input
+    }
+
+    return _from(ptr, 1);
+  }
+
+  function from(SchemaType[] memory input) internal pure returns (bytes memory output) {
+    bytes32 ptr;
+    assembly {
+      ptr := input
+    }
+
+    return _from(ptr, 1);
+  }
+
+  function from(uint16[] memory input) internal pure returns (bytes memory output) {
+    bytes32 ptr;
+    assembly {
+      ptr := input
+    }
+
+    return _from(ptr, 2);
+  }
+
+  function from(uint32[] memory input) internal pure returns (bytes memory output) {
+    bytes32 ptr;
+    assembly {
+      ptr := input
+    }
+
+    return _from(ptr, 4);
+  }
+
+  function from(bytes24[] memory input) internal pure returns (bytes memory output) {
+    bytes32 ptr;
+    assembly {
+      ptr := input
+    }
+
+    return _from(ptr, 24, true);
+  }
+
+  function from(string memory input) internal pure returns (bytes memory output) {
+    assembly {
+      output := input
+    }
+  }
+
+  function from(address[] memory input) internal pure returns (bytes memory output) {
+    bytes32 ptr;
+    assembly {
+      ptr := input
+    }
+
+    return _from(ptr, 20);
+  }
+
+  function _from(bytes32 _ptr, uint256 _bytesPerElement) internal pure returns (bytes memory output) {
+    return _from(_ptr, _bytesPerElement, false);
+  }
+
+  function _from(
+    bytes32 _ptr,
+    uint256 _bytesPerElement,
+    bool leftAligned
+  ) internal pure returns (bytes memory output) {
+    uint256 shiftBits = leftAligned ? 0 : (32 - _bytesPerElement) * 8;
+
+    assembly {
+      let inputLength := mload(_ptr)
+      let outputBytes := mul(inputLength, _bytesPerElement)
+
+      // Allocate memory for the output and store its length
+      output := mload(0x40)
+      mstore(output, outputBytes)
+
+      // Update the free memory pointer
+      mstore(0x40, add(output, add(32, outputBytes)))
+
+      let outputPtr := add(output, 32)
+      for {
+        let inputPtr := add(_ptr, 32) // Start at first element
+        // Stop at last element
+      } lt(inputPtr, add(add(_ptr, 32), mul(mload(_ptr), 32))) {
+        inputPtr := add(inputPtr, 32) // Go to next input element
+        outputPtr := add(outputPtr, _bytesPerElement) // Go to next output slot
+      } {
+        mstore(outputPtr, shl(shiftBits, mload(inputPtr))) // Store the value in minimal bytes
+      }
+    }
+  }
+
+  // Needs unique name to avoid conflict with `from(uint32)`
+  function fromUint8(uint8 input) internal pure returns (bytes memory output) {
+    return abi.encodePacked(input);
```

Comment:
seems easier to just use `abi.encodePacked` instead of having 96 converter methods; I guess this is a leftover after `bytes.concat`?